modules/tensor_mechanics/src/materials/HillElastoPlasticityStressUpdate.C

```diff
@@ -61,12 +73,101 @@ HillElastoPlasticityStressUpdateTempl<is_ad>::HillElastoPlasticityStressUpdateTe
                                                                                   "alpha_matrix")),
     _sigma_tilde(this->template declareGenericProperty<DenseVector<Real>, is_ad>(this->_base_name +
                                                                                  "sigma_tilde")),
+    _sigma_tilde_rotated(this->template declareGenericProperty<DenseVector<Real>, is_ad>(
+        this->_base_name + "sigma_tilde_rotated")),
     _hardening_derivative(0.0),
     _yield_condition(1.0),
     _yield_stress(this->template getParam<Real>("yield_stress")),
     _hill_tensor(this->template getMaterialPropertyByName<DenseMatrix<Real>>(this->_base_name +
-                                                                             "hill_tensor"))
+                                                                             "hill_tensor")),
+    _rotation_matrix(
+        this->template declareProperty<RankTwoTensor>(this->_base_name + "rotation_matrix")),
+    _rotation_matrix_transpose(this->template declareProperty<RankTwoTensor>(
+        this->_base_name + "rotation_matrix_transpose")),
+    _rotation_matrix_old(
+        this->template getMaterialPropertyOld<RankTwoTensor>(this->_base_name + "rotation_matrix")),
+    _rotation_matrix_transpose_old(this->template getMaterialPropertyOld<RankTwoTensor>(
+        this->_base_name + "rotation_matrix_transpose")),
+    _local_cylindrical_csys(this->template getParam<bool>("local_cylindrical_csys")),
+    _axis(this->template getParam<MooseEnum>("axis").template getEnum<Axis>())
 {
+  if (_local_cylindrical_csys && !parameters.isParamSetByUser("axis"))
+  {
+    this->paramWarning(
+        "local_cylindrical_csys",
+        "\nIf parameter local_cylindrical_csys is provided then parameter axis should be also "
+        "provided. Continuing with default value of axis=1 (y-axis)");
```

Comment:
See if it makes sense to make this an error. It seems we can in good faith force the user to be thorough in the problem definition.

Reply:
Changed paramWarning to paramError.